src/cgame/rocket/emojis/generate_emoji_map.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python
+
+import requests
+import sys
+import os
+
+EMOJI_MAP_SRC = 'https://raw.githubusercontent.com/muan/emojilib/main/dist/emoji-en-US.json'
+GOOGLE_EMOJI_SRC = 'https://github.com/googlefonts/emoji-metadata/raw/main/emoji_15_0_ordering.json'
+
+CPLUSPLUS_CODE_HEADER="""
+#include "emojimap.h"
+
+#include <unordered_map>
+
+const std::unordered_map<Str::StringRef, const char*> emojimap = {
+"""
+
+CPLUSPLUS_CODE_FOOTER="""};
+
+Str::StringRef FindEmoji( Str::StringRef emoji )
+{
+	auto it = emojimap.find( emoji.c_str() );
+	if ( it == emojimap.end() )
+	{
+		return "";
+	}
+	return it->second;
+}
+
+"""
+
+def render_base(base):
+    s = ''
+    for codept in base:
+        s += chr(codept)
+    return s
+
+def escapeq(s):
+    return s.replace('"', '\\"').replace('\\', '\\\\')
```

Comment:
This looks wrong

Reply:
How so? `s.replace()` returns a string which as a method `replace`. There probably is a better way of doing this, but this isn't super performance critical.

Reply:
oic. fixed